source/simulator/core.cc

```diff
@@ -1503,15 +1504,25 @@ namespace aspect
       typedef std::set< std::pair< std::pair< types::boundary_id, types::boundary_id>, unsigned int> >
       periodic_boundary_set;
       periodic_boundary_set pbs = geometry_model->get_periodic_boundary_pairs();
-
-      for (periodic_boundary_set::iterator p = pbs.begin(); p != pbs.end(); ++p)
-        {
-          DoFTools::make_periodicity_constraints(dof_handler,
-                                                 (*p).first.first,  // first boundary id
-                                                 (*p).first.second, // second boundary id
-                                                 (*p).second,       // cartesian direction for translational symmetry
-                                                 constraints);
-        }
+      
+      const double periodicity_factor = 1.;
+
+      std::vector<GridTools::PeriodicFacePair<typename DoFHandler<dim>::cell_iterator> >
+      matched_pairs;
+      FullMatrix<double> rotation_matrix(dim); 
+      rotation_matrix[0][1] = 1.;
+      rotation_matrix[1][0] = -1.;
+
+      GridTools::collect_periodic_faces
+        ( dof_handler, /*b_id1*/ 2, /*b_id2*/ 3,
+           /*direction*/ 0, matched_pairs, 
```

Comment:
Here you use 0, above you use 1. Not sure which one is correct.

Reply:
Changed to 0 for now so all match